src/errors/SiteLaunchError.ts

```diff
@@ -0,0 +1,11 @@
+import { BaseIsomerError } from "@root/errors/BaseError"
+
+export default class SiteLaunchError extends BaseIsomerError {
+  constructor(siteName: string) {
+    super({
+      status: 500,
+      code: "InfraError",
+      message: `The given file: ${siteName} was not a config file!`,
```

Comment:
@seaerchin @kishore03109 should we starting writing new errors as per V2 format similar to `CollectionPageService`?

example
```
 const err = new BadRequestError(
        `Special characters not allowed when renaming files. Given name: ${newFileName}`,
        { fileName: newFileName },
        ComponentTypes.Service,
        FileCodes.CollectionPageService
      )
      err.isV2Err = true
      throw err
    }
```

if so things like `meta` can contain the `siteName` and if it is compatible with the new interface it can expose the new methods in `IsomerError.ts`

```
/* Namespace for Error functions */
export const IsomerError = {
  toExternalRepresentation: (e: IsomerInternalError): IsomerExternalError => ({
    name: e.code,
    code: getExternalCode(e.componentCode, e.fileCode, e.code),
    message: e.message,
    isV2Err: true,
  }),
  getLog: (error: IsomerInternalError): Record<string, unknown> => ({
    ...error,
  }),
}

```

Reply:
yes, but not inclined to do it here. this PR is already well scoped and the errors don't directly impact functionality. i think we should not let perfect be the enemy of good and as long as the PR contains good code and has relevant test coverage, should be good to go. 

the error can go in as part of another ticket.

Reply:
ea88911320ab365c68528e1a4b6b66038db942b0
following the inspiration from `BadRequestError` from `CollectionPageService`